services/migrations/migrate.go

```diff
@@ -312,9 +314,9 @@ func migrateRepository(doer *user_model.User, downloader base.Downloader, upload
 		}
 
 		// Once all releases (if any) are inserted, sync any remaining non-release tags
-		if err = uploader.SyncTags(); err != nil {
-			return err
-		}
+		// if err = uploader.SyncTags(); err != nil {
+		// 	return err
+		// }
```

Comment:
It seems this code is causing issues with release syncs. We got all the releases already and then tries to sync "any remaining non-release tags", which is not the case.

It's actually syncing all the tags including release tags by calling `SyncReleasesWithTags`, and even worse, it's deleting ALL the releases and recreating them in a broken way:

If repo is a mirror, `pullMirrorReleaseSync` in `modules/repository/repo.go` is called. Then all releases are deleted, and releases are recreated from tags without any `Author`, `NumCommits`, etc. This is the wrong behavior because we do need the release information.

```go
for _, tag := range tags {
	release := repo_model.Release{
		RepoID:       repo.ID,
		TagName:      tag.Name,
		LowerTagName: strings.ToLower(tag.Name),
		Sha1:         tag.Object.String(),
		// NOTE: ignored, since NumCommits are unused
		// for pull-mirrors (only relevant when
		// displaying releases, IsTag: false)
		NumCommits:  -1,
		CreatedUnix: timeutil.TimeStamp(tag.Tagger.When.Unix()),
		IsTag:       true,
	}
	if err := db.Insert(ctx, release); err != nil {
		return fmt.Errorf("unable insert tag %s for pull-mirror Repo[%d:%s/%s]: %w", tag.Name, repo.ID, repo.OwnerName, repo.Name, err)
	}
}
```